src/lib/debouncePromise.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import debounce from 'lodash/debounce';
+
+function debouncePromise<T>(
+  fn: (...args: any[]) => Promise<T>,
+  delay: number
+): (...args: any[]) => Promise<T> {
+  let promise;
```

Comment:
-1: Name is really bad, this is never a promise. Also not too happy with the order of calls/assignments, maybe you can move the promiseHolder `promise` down, and pass it as an argument into the debounce function? Would have been easier to read for me. Typing for this would also be nice.

Reply:
The function debounces (verb) a promise by a given delay. Thats why I named it like this. What do you think would fit better?

Good idea of passing the promise into the debounce function. Will fix that.

Reply:
Passing the promise into the debounce function does not work as we need to set the promise holder `promise` to null once the debounce function is finally called. 😕 

The promise holder is also automatically typed by flow due to its following usage inside the function.

<img width="662" alt="screenshot 2018-10-15 17 15 12" src="https://user-images.githubusercontent.com/1609692/46960086-f85baa80-d09d-11e8-8b3f-e0b84785349d.png">

Reply:
I was talking about the local variable `promise`, which is not a promise.

Reply:
Ahh … sorry. Now I see. What do you think about `promiseHolder` or `promiseCache?

Reply:
Good point - I did not think about the resetting. Either name is fine :)